### ISSUE TITLE
framework referring to XCTest reference does not link in app

### ISSUE THREAD

**Issue by kennebecstars** [Carthage/Carthage#1407]
My question is why is Xcode 7.3.1 not able to run in the simulator? I get a message out of lldb and we stop the run:

dyld: Library not loaded: @rpath/XCTest.framework/XCTest
  Referenced from: ~/Library/Developer/CoreSimulator/Devices/2796378A-600E-4FB0-93C4-AAA24FD3E72C/data/Containers/Bundle/Application/607C7701-D92E-4331-89EA-44567E943D2A/MyApp.app/MyApp
  Reason: image not found
(lldb) 

What do I do? ***Note: I have made the XCTest.framework required in the build phase.

* `carthage version`: 0.11.0
* `xcodebuild -version`: 7.3.1
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? No
* Are you using `--use-submodules`? No

**Cartfile**
```
github "kif-framework/KIF" ~> 3.4.1
github "erikdoe/ocmock"
github "hamcrest/OCHamcrest"
github "specta/specta" "master"
github "specta/expecta"
```

**Carthage Output**
```
$ carthage update
*** Fetching expecta
*** Fetching specta
*** Fetching OCHamcrest
*** Fetching ocmock
*** Fetching KIF
*** Checking out KIF at "v3.4.2"
*** Downloading expecta.framework binary at "v1.0.5"
*** Downloading ocmock.framework binary at "OCMock 3.3.1"
*** Downloading OCHamcrest.framework binary at "v5.4.0"
*** Checking out OCHamcrest at "v5.4.0"
*** Checking out specta at "fab66106480b34a8e4341eee360d2c31d37ce015"
*** xcodebuild output can be found in /var/folders/5c/vdn3cxs96q1gf22wkyqys_6m0000gn/T/carthage-xcodebuild.gInbO3.log
*** Building scheme "KIFFramework" in Testable.xcworkspace
*** Building scheme "OCHamcrest" in OCHamcrest.xcodeproj
*** Building scheme "OCHamcrest-watchOS" in OCHamcrest.xcodeproj
*** Building scheme "OCHamcrest-tvOS" in OCHamcrest.xcodeproj
*** Building scheme "OCHamcrest-iOS" in OCHamcrest.xcodeproj
*** Building scheme "Specta" in Specta.xcworkspace
*** Building scheme "Specta-iOS" in Specta.xcworkspace
./Carthage/Carthage/Checkouts/OCHamcrest/Source/Library/Text/HCIsEqualCompressingWhiteSpace.h:43:85: warning: GCC does not allow 'deprecated' attribute in this position on a function definition [-Wgcc-compat]
./Carthage/Carthage/Checkouts/OCHamcrest/Source/Library/Text/HCIsEqualCompressingWhiteSpace.h:43:85: warning: GCC does not allow 'deprecated' attribute in this position on a function definition [-Wgcc-compat]
./Carthage/Carthage/Checkouts/OCHamcrest/Source/Library/Text/HCIsEqualCompressingWhiteSpace.h:43:85: warning: GCC does not allow 'deprecated' attribute in this position on a function definition [-Wgcc-compat]
./Carthage/Carthage/Checkouts/OCHamcrest/Source/Library/Text/HCIsEqualCompressingWhiteSpace.h:43:85: warning: GCC does not allow 'deprecated' attribute in this position on a function definition [-Wgcc-compat]
./Carthage/Carthage/Checkouts/OCHamcrest/Source/Library/Text/HCIsEqualCompressingWhiteSpace.h:43:85: warning: GCC does not allow 'deprecated' attribute in this position on a function definition [-Wgcc-compat]
./Carthage/Carthage/Checkouts/OCHamcrest/Source/Library/Text/HCIsEqualCompressingWhiteSpace.h:43:85: warning: GCC does not allow 'deprecated' attribute in this position on a function definition [-Wgcc-compat]
./Carthage/Carthage/Checkouts/OCHamcrest/Source/Library/Text/HCIsEqualCompressingWhiteSpace.h:43:85: warning: GCC does not allow 'deprecated' attribute in this position on a function definition [-Wgcc-compat]
./Carthage/Carthage/Checkouts/OCHamcrest/Source/Library/Text/HCIsEqualCompressingWhiteSpace.h:43:85: warning: GCC does not allow 'deprecated' attribute in this position on a function definition [-Wgcc-compat]
./Carthage/Carthage/Checkouts/OCHamcrest/Source/Library/Text/HCIsEqualCompressingWhiteSpace.h:43:85: warning: GCC does not allow 'deprecated' attribute in this position on a function definition [-Wgcc-compat]
$
```

**Comment by mdiep** [comment]
Why are you trying to link XCTest to your app directly? Normally it'd just be a part of the unit test bundle.

**Comment by kennebecstars** [comment]
It is set as optional. I want to test UI functionality using KIF framework and these settings have worked in Xcode 7.3, but are failing now in Xcode 7.3.1.

**Comment by kennebecstars** [comment]
Thanks, @mdiep for looking at this. With experience at carthage, as well, I hope you know what's wrong. Should I just remove the KIF and XCTest framework entries from the build. KIF setup asks that we do this but remove these before final deployment of the release build.

**Comment by mdiep** [comment]
> ```
> dyld: Library not loaded: @rpath/XCTest.framework/XCTest
> ```

It seems like you need to adjust your `@rpath` to include the path containing `XCTest.framework`. But I'm not sure. This is really a KIF issue, not a Carthage one.

**Comment by kennebecstars** [comment]
I removed the references and all is working, again. So I'm chalking this issue up to fixes made in 7.3.1 from 7.3. Regression, regression, regression. Thanks, @mdiep!

**Comment by kennebecstars** [comment]
Already updated and closed question. Thank you!
